core/src/main/java/org/apache/calcite/adapter/enumerable/RexImpTable.java

```diff
@@ -609,6 +608,8 @@ public static CallImplementor createImplementor(
           final List<Expression> expressions =
               translator.translateList(call2.getOperands(), nullAs);
           return Expressions.foldAnd(expressions);
+        case NOT_POSSIBLE:
+          // This doesn't mean that none of the arguments might be null, ex: (s and s is not null)
```

Comment:
I wonder what changes if we use `case NOT_POSSIBLE: nullAs=TRUE;` instead and keep this `case NOT_POSSIBLE` on its original place.